modules/camera/base_camera.py

```diff
@@ -0,0 +1,28 @@
+"""
+Base class for camera device
+"""
+
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+class BaseCameraDevice(ABC):
+    """
+    Abstract class for camera device implementations.
+    TODO: could leverage the abc library's tags more (required properties/methods/etc.)
```

Comment:
Nevermind, just read up on it and the abstractproperty decorator is deprecated since python 3.3

We can just use the @property decorator. However, I believe camera is supposed to be "private"